### PR TITLE
Windows: Create a QT 6.6.0-Next task 

### PR DESCRIPTION
**Pull request by strseb**
## Description

We already have a[ android-next task ](https://treeherder.mozilla.org/jobs?repo=mozilla-vpn-client&searchStr=next) that builds the client using a more recent version of qt. 
Given we want to upgrade windows too soon, let's create them here aswell.

### CODE REVIEW

**Review by brizental — APPROVED** [review]

File: taskcluster/scripts/toolchain/compile_qt_6.ps1 [hunk 1]
```diff
@@ -74,6 +83,75 @@ Set-Location $FETCHES_PATH/qt-everywhere-src-$QT_VERSION
   -qt-zlib  `
   -openssl-runtime `
   -prefix $BUILD_PREFIX `
+} else {
+  # For newer qt versions, let's trim what we dont need.
```
Comment (vinocher) [inline taskcluster/scripts/toolchain/compile_qt_6.ps1:87]: A comment that explains where these options are documented would be helpful to the reader. I found some in qtbase\config_help.txt.

File: taskcluster/scripts/toolchain/compile_qt_6.ps1 [hunk 2]
```diff
@@ -74,6 +83,75 @@ Set-Location $FETCHES_PATH/qt-everywhere-src-$QT_VERSION
...
+  ./configure.bat `
+  -static  `
+  -opensource  `
+  -debug-and-release `
+  -confirm-license  `
+  -silent  `
+  -make libs  `
+  -no-feature-dynamicgl `
```
Comment (vinocher) [inline taskcluster/scripts/toolchain/compile_qt_6.ps1:95]: Should we use the following which were used in 6.2?
  -nomake tests  `
  -nomake examples  `

File: taskcluster/scripts/build/windows_clang_cl.ps1 [hunk 3]
```diff
@@ -24,9 +24,12 @@ Copy-Item -Path $env:VCToolsRedistDir\\MergeModules\\Microsoft_VC143_CRT_x86.msm
 
 
 # Setup Openssl Import
-$SSL_PATH = resolve-path "$FETCHES_PATH/QT_OUT/SSL"
-$env:OPENSSL_ROOT_DIR = (resolve-path "$SSL_PATH").toString()
-$env:OPENSSL_USE_STATIC_LIBS = "TRUE"
+$SSL_PATH = "$FETCHES_PATH/QT_OUT/SSL"
+if (Test-Path -Path $SSL_PATH) {
```
Comment (vinocher) [inline taskcluster/scripts/build/windows_clang_cl.ps1:28]: Should we also remove the openssl binaries from the windows\installer\MozillaVPN.wxs?
        <File Source="../../libcrypto-1_1-x64.dll" />
        <File Source="../../libssl-1_1-x64.dll" />

**Review by gabrielBusta — APPROVED** [review]

File: taskcluster/scripts/build/windows_clang_cl.ps1 (hunk 3 above)
Comment (strseb) [reply taskcluster/scripts/build/windows_clang_cl.ps1:28]: Yes, once we will ship with that version. This build's task is to make sure we have build's with the next version of qt ready and passable to qa at any point in time :)

File: taskcluster/scripts/toolchain/compile_qt_6.ps1 (hunk 1 above)
Comment (strseb) [reply taskcluster/scripts/toolchain/compile_qt_6.ps1:87]: That's great! Added more context on how i drilled to that list

File: taskcluster/scripts/toolchain/compile_qt_6.ps1 [hunk 4]
```diff
@@ -46,7 +51,11 @@ $BUILD_PREFIX = (resolve-path "$REPO_ROOT_PATH/QT_OUT").toString()
 # Enter QT source directory
 Set-Location $FETCHES_PATH/qt-everywhere-src-$QT_VERSION
 
-./configure.bat `
+$ErrorActionPreference = "Stop"
+
+if($QT_VERSION_MAJOR -eq "6.2" ){
+  # We should not chane the behavior mid release.
```
Comment (vinocher) [inline taskcluster/scripts/toolchain/compile_qt_6.ps1:57]: Nit: Typo for "change"

**Review by vinocher — APPROVED** [review]

**Pull request merged**